ddtrace/llmobs/_integrations/langchain.py

```diff
@@ -234,3 +236,29 @@ def record_usage(self, span: Span, usage: Dict[str, Any]) -> None:
         total_cost = span.get_metric(TOTAL_COST)
         if total_cost:
             self.metric(span, "incr", "tokens.total_cost", total_cost)
+
+    def format_io(
+        self,
+        messages,
+    ):
+        """
+        Formats input and output messages for serialization to JSON.
+        Specifically, makes sure that any schema messages are converted to strings appropriately.
+        """
+        if isinstance(messages, dict):
+            formatted = {}
+            for key, value in messages.items():
+                if isinstance(value, list):
+                    formatted[key] = [self.format_io(item) for item in value]
+                else:
+                    formatted[key] = self.format_io(value)
+            return formatted
+        if isinstance(messages, list):
+            return [self.format_io(message) for message in messages]
+        return self.get_content_from_message(messages)
+
+    def get_content_from_message(self, message) -> str:
+        """Extracts the content from a message (AIMessage, HumanMessage, SystemMessage) object."""
+        if isinstance(message, str):
+            return message
+        return message.__dict__.get("content", "")
```

Comment:
do we want to also include the role here too? so it would look something like
```
("user", "hey there!")
```
instead of just
```
"hey there!"
```

or, do we just want to stringify it entirely, so it would _just_ look like:
```
"content='hey there!'"
```

Reply:
cool i can add that in

Reply:
specifying the role for the langchain LLM call already works correctly:
<img width="675" alt="Screenshot 2024-05-08 at 10 29 24 AM" src="https://github.com/DataDog/dd-trace-py/assets/106700075/3aacf2f7-3648-417f-84db-1894380384b3">